app/change_set_persisters/change_set_persister.rb

```diff
@@ -164,8 +164,10 @@ def add
       def run; end
     end
 
-    def with(metadata_adapter:)
-      yield self.class.new(metadata_adapter: metadata_adapter, storage_adapter: storage_adapter, transaction: true, characterize: @characterize, queue: queue, handlers: handlers)
+    def with(metadata_adapter: self.metadata_adapter, storage_adapter: self.storage_adapter)
```

Comment:
I'd never noticed this method before. It's an interesting kind-of-a-factory? Could you add a comment explaining its use?